igor-web/src/main/groovy/com/netflix/spinnaker/igor/travis/TravisBuildMonitor.groovy

```diff
@@ -92,6 +100,19 @@ class TravisBuildMonitor implements PollingMonitor{
                 }
             } as Action0, 0, pollInterval, TimeUnit.SECONDS
         )
+
+        worker.schedulePeriodically(
+            {
+                if (isInService()) {
```

Comment:
I've had it running for 9 hours with this code now without failure. I can add a semaphore, but then it might just as well be the changedBuilds method that calls repositorySync every 10th time it runs or something. What do you think?
1. Leave it as is.
2. Introduce semaphore.
3. Let changedBuilds call repositorySync for every X time it runs.